### ISSUE TITLE
CA1001 should say which fields are disposable

### ISSUE THREAD

**Issue by pharring** [dotnet/roslyn-analyzers#1564]
#### Analyzer package

Example: Microsoft.CodeQuality.CSharp.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.CSharp.Analyzers/2.6.0)

#### Analyzer

CA1001: Types that own disposable fields should be disposable

#### Repro steps

Analyze this:
```C#
using System.IO;

namespace ClassLibrary1
{
    public class Class1
    {
        private readonly Stream _stream1 = new MemoryStream();
        private readonly Stream _stream2 = new MemoryStream();
        private readonly Stream _stream3 = new MemoryStream();
        private readonly FileInfo _fileInfo = new FileInfo(@"C:\test.txt");
    }
}

```

This reports CA1001 correctly, but I'm left to figure out on my own which fields are disposable. Even if I implement Dispose correctly, I have no confidence that I've called Dispose on all the disposable fields.

My suggestion is to augment CA1001's message with the name(s) of the disposable field(s).

#### Expected behavior
warning CA1001: Type 'Class1' owns disposable fields (_stream1, stream2, _stream3) but is not disposable

#### Actual behavior
warning CA1001: Type 'Class1' owns disposable fields but is not disposable

**Comment by sharwell** [comment]
Do we have an analyzer that tracks the individual fields after the Dispose method is added? If so, it seems like the information would be provided there (ideally live in the editor).

**Comment by pharring** [comment]
Looks like [CA2213](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2213-disposable-fields-should-be-disposed) might be it.

**Comment by mavasani** [comment]
We have decided to implement CA2213 - https://github.com/dotnet/roslyn-analyzers/pull/1595. We should be able to extend CA1001 to compute disposable fields in a type.

**Comment by mavasani** [comment]
It seems that CA1001 already has a local set of disposable fields - we just to need to update the error message to display the field names. We should also improve the preciseness of this rule using DFA DisposeAnalysis used by other rules.
